src/openfermion/resource_estimates/molecule/df_lrps_utils.py

```diff
@@ -0,0 +1,195 @@
+# Code for implementing the DF + LRPS decomposition found in https://arxiv.org/abs/2409.18277
+#
+# Notes: 
+#     1. All tensors are assumed to be indexed over spatial orbitals
+#     2. g_{ijkl} tensor corresponds to \sum_{\sigma, tau} a_{i\sigma}^\dagger a_{j\sigma} a_{k\tau}^\dagger a_{l\tau} terms in electronic Hamiltonian
+#     3. All functions assume that g_{(ij),(kl)} matrix is positive-semidefinite.
+#
+
+import numpy as np
+from itertools import product
+
+# First set of functions is for calculating the 1-norm of DF decompositions
+
+def l1norm_onebody_spatial(obt):
+    """
+    return 1-norm of LCU decomposition of 1e Hamiltonian with obt as one-body-tensor
+
+    Args:
+        obt (ndarray): one-body-tensor of a one-electron Hamiltonian
+
+    Returns:
+        1-norm: 1-norm of associated fermionic LCU decomposition
+    """
+    return np.sum(np.abs(np.linalg.eigh(obt)[0]))
+
+def l1norm_onebody_NSS_spatial(obt):
+    """
+    return 1-norm of LCU decomposition of symmetry shifted 1e Hamiltonian with obt as one-body-tensor
+    the optimal symmetry shift parameter for 1-norm minimization is used (median of obt_fu eigenvalues)
+
+    Args:
+        obt (ndarray): one-body-tensor of a one-electron Hamiltonian
+
+    Returns:
+        1-norm: 1-norm of associated fermionic LCU decomposition
+    """
+    eps = np.linalg.eigh(obt)[0]
+    shift = np.median(eps)
+    return np.sum(np.abs(eps - shift))
+
+def l1norm_lr_twobody_spatial(eps_params):
+    """
+    return the 1-norm of two-body-tensor which is decomposed into "sum of squares" fragments. The fragments are assumed to have the form
+    U^\dagger (\sum_{i\sigma} \epsilon_i \hat{n}_{i\sigma}) U. 
+
+    Args:
+        eps_params (list of ndarray): list of epsilon parameters for each low-rank fragment 
+
+    Returns:
+        l1norm: 1-norm of associated DF decomposition
+    """
+    l1norm = 0
+    for alpha in range(len(eps_params)):
+        l1norm += 0.5 * (np.sum(np.abs(eps_params[alpha])))**2
+    return l1norm
+
+# second set of functions is for calculating the DF decomposition. 
+
+def obtain_lowrank_decomposition(tbt):
```

Comment:
hardcoded cutoffs. 1) can you use existing openfermion infrastructure 2) make cutoffs inputs or optional inputs